hydroflow_datalog_core/src/lib.rs

```diff
@@ -24,16 +27,21 @@ pub fn parse_pipeline(
     literal: &proc_macro2::Literal,
     offset: usize,
 ) -> Result<Pipeline, Vec<Diagnostic>> {
-    syn::parse_str(&code_str.value).map_err(|err| {
-        let (start, end) = code_str.span;
-        // TODO(mingwei): incorporate `err.span()` somehow
-        let subspan = literal.subspan(start + offset..end + offset);
-        vec![Diagnostic {
-            span: subspan.unwrap_or(Span::call_site()),
-            level: Level::Error,
-            message: format!("Failed to parse input pipeline: {}", err),
-        }]
-    })
+    let (start, end) = code_str.span;
+    // TODO(mingwei): incorporate `err.span()` somehow
+    let subspan = literal
+        .subspan(start + offset..end + offset)
+        .unwrap_or(Span::call_site());
+
+    syn::LitStr::new(&code_str.value, subspan)
+        .parse()
+        .map_err(|err| {
+            vec![Diagnostic {
+                span: subspan,
+                level: Level::Error,
+                message: format!("Failed to parse input pipeline: {}", err),
+            }]
+        })
```

Comment:
This ought to use `get_span` too? Guess there wouldn't be a way to incorporate `err.span()` though then

Also need to move that commend down to line 39